orderbook/src/database/orders.rs

```diff
@@ -202,7 +202,31 @@ struct OrdersQueryRow {
 }
 
 impl OrdersQueryRow {
+    fn calculate_status(&self) -> OrderStatus {
+        match self.kind {
+            DbOrderKind::Buy => {
+                if !self.sum_buy.is_zero() && self.sum_buy == self.buy_amount {
+                    return OrderStatus::Fulfilled;
+                }
+            }
```

Comment:
Do we need to be concerned with any epsilons here for classifying a partially fillable order as fully filled? Something like |x-y| < 10^9. Just wondering. Same question for sell orders.

Reply:
I did just that in the explorer https://github.com/gnosis/gp-ui/blob/develop/src/utils/operator.ts#L21

But here, I didn't because I'm following what's used in the DB filter https://github.com/gnosis/gp-v2-services/blob/main/orderbook/src/database/orders.rs#L173

Have nothing against adding the epsilon here as well.

Reply:
Let's see what others think and whether or not we are even going to include partially filled. I do think though, that if you separated the logic of filledness and openness into a tuple, that the id statements (I.e. the code) will wind up a lot cleaner and easier to maintain for future devs. I myself have gotten into some pretty messy code when it comes to having a lot of independent statuses all in one go.

Reply:
Addressed on https://github.com/gnosis/gp-v2-services/pull/715/commits/9a1c82999a287eba9f3aa9b09503babbdc85322c

Reply:
I am not sure about this. I don't find it more intuitive that there is an epsilon for partially fillable orders. You could also say that if it was filled once it is filled or it is only filled if the amount is the real full amount. Or we could add a new status PartiallyFilled that is set until it is fully Filled. I would stay with one of those solutions until we get actual experience with partially fillable orders and how they would be displayed in the frontend etc.

Reply:
Based on @vkgnosis  and @fleupold (https://github.com/gnosis/gp-v2-services/pull/715#discussion_r651671530) comments, I'll revert the change and only set it to filled when 100% filled.

Regarding the partially filled status, see my concerns with a dedicated status and a proposal on https://github.com/gnosis/gp-v2-services/pull/715#discussion_r651163943

Reply:
As discussed over slack, will keep it without a dedicated flag for the partial fill for now.